lib/logger/lib/logger.ex

```diff
@@ -750,8 +749,25 @@ defmodule Logger do
       Logger.warn(fn -> {"dynamically calculated warning", [additional: :metadata]} end)
 
   """
+  # TODO: Deprecate it in favour of `warning/1-2` macro
   defmacro warn(chardata_or_fun, metadata \\ []) do
-    maybe_log(:warn, chardata_or_fun, metadata, __CALLER__)
+    maybe_log(:warning, chardata_or_fun, metadata, __CALLER__)
+  end
+
+  @doc """
+  Logs a warning message.
+
+  Returns `:ok` or an `{:error, reason}` tuple.
+
+  ## Examples
+
+      Logger.warning("knob turned too far to the right")
+      Logger.warning(fn -> "dynamically calculated warning" end)
+      Logger.warning(fn -> {"dynamically calculated warning", [additional: :metadata]} end)
+
+  """
+  defmacro warning(chardata_or_fun, metadata \\ []) do
+    maybe_log(:warning, chardata_or_fun, metadata, __CALLER__)
   end
```

Comment:
Let's not add this one for now (since we are not adding `Logger.log(:warning, ...)`. Once this is merged, we will create an issue so we don't forget to add it for 1.11.